student/src/Form/StudentSignUpForm.php

```diff
@@ -0,0 +1,177 @@
+<?php
+ 
+ /**
+  * Displays link if student is logged in and not signed up for course.
+  * php version 7.2.24
+  * 
+  * @category Custom_Form
+  * @package  Student
+  * @author   Neil Bridges <neil@discoverygarden.com>
+  * @license  test.com  test
+  * @link     test
+  * @return   /custom/student/src/Form/StudentSignUpForm.php
+  */  
+ 
+namespace Drupal\student\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Implements an example form.
+ *    
+ * Build Form Description
+ *
+ * @category Student
+ * @package  Student
+ * @author   Neil Bridges <neil@discoverygarden.com>
+ * @license  student
+ * @link     Student 
+ * @return   /custom/student/src/Form/StudentSignUpForm.php      
+ */ 
+class StudentSignUpForm extends FormBase
+{
+        
```

Comment:
It's useful to have doc comments on all variables and have them type spaced. So for example, if the account is containing a AccountProxy class, it should appear like this:

```suggestion
/**
 * Account Proxy object.
 *
 * @var \Drupal\Core\Session\AccountProxy
 */
 protected $account;
```